kvstore/memory_test.go

```diff
@@ -0,0 +1,234 @@
+// Copyright 2021 Contributors to the Veraison project.
+// SPDX-License-Identifier: Apache-2.0
+package kvstore
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	testKey    = `psa://tenant-1/deadbeef/beefdead`
+	testVal    = `{"some": "json"}`
+	altTestKey = `psa://tenant-2/cafecafe/cafecafe`
+	altTestVal = `[1, 2, 3]`
+)
+
+func TestMemory_Init_missing_store_type(t *testing.T) {
+	s := Memory{}
+
+	cfg := map[string]interface{}{
+		"some-random-directive": "blabla",
+	}
+
+	expectedErr := `missing "store-type" directive`
+
+	err := s.Init(cfg)
+	assert.EqualError(t, err, expectedErr)
+}
+
+func TestMemoryStore_Init_empty_store_type(t *testing.T) {
+	s := Memory{}
+
+	cfg := map[string]interface{}{
+		"store-type": "",
+	}
+
+	expectedErr := `invalid "store-type": unknown type ""`
+
+	err := s.Init(cfg)
+	assert.EqualError(t, err, expectedErr)
+}
+
+func TestMemoryStore_Init_unknown_store_type(t *testing.T) {
+	s := Memory{}
+
+	cfg := map[string]interface{}{
+		"store-type": "some-random-type",
+	}
+
+	expectedErr := `invalid "store-type": unknown type "some-random-type"`
+
+	err := s.Init(cfg)
+	assert.EqualError(t, err, expectedErr)
+}
+
+func TestMemoryStore_Init_bad_store_type(t *testing.T) {
```

Comment:
Tests on Line 32, 45 and 58 can be potentially combined into a single table driven test.

Reply:
True, and others as well in this same lot could be made table-driven, I think.  Question: is it necessary?

Reply:
Well, only benefit I see is the code compactness and logical coherence! Reader sees all negative tests in one place.

I leave this up to you on your best judgement!

Reply:
On compactness I agree, logical coherence not so much.  Anyway, if you really prefer this way, I'll do it.

Reply:
Addressed in 0dfbb0e